test-util/src/main/java/io/camunda/zeebe/test/util/bpmn/random/blocks/SubProcessBlockBuilder.java

```diff
@@ -115,10 +115,12 @@ public ExecutionPathSegment generateRandomExecutionPath(final ExecutionPathConte
     final var internalExecutionPath = embeddedSubProcessBuilder.findRandomExecutionPath(context);
 
     if (!hasBoundaryEvents || !internalExecutionPath.canBeInterrupted() || random.nextBoolean()) {
-      result.append(internalExecutionPath, true);
+      final var isDifferentFlowScope = true;
+      result.append(internalExecutionPath, isDifferentFlowScope);
     } else {
       internalExecutionPath.cutAtRandomPosition(random);
-      result.append(internalExecutionPath, true);
+      final var isDifferentFlowScope = true;
+      result.append(internalExecutionPath, isDifferentFlowScope);
```

Comment:
I guess this is the part where we force appending the execution path to the block (i.e. append it even if a terminate end event has been reached already). What does this part here add that must be executed even when a terminate end event has been reached?